src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java

```diff
@@ -239,4 +238,102 @@ private void updateOrInsertAllCodeSystems(List<CodeSystem> codeSystemList) {
       }
     }
   }
+
+  public List<Code> retrieveCodesList(List<Map<String, String>> codeList, String apiKey) {
+    return codeList.stream()
+        .map(
+            code -> {
+              List<CodeSystemEntry> codeSystemEntries = mappingService.getCodeSystemEntries();
+              String codeName = code.get("code");
+              String codeSystemName = code.get("codeSystem");
+              String oid = code.get("oid") != null ? code.get("oid").replaceAll("'|'", "") : null;
+
+              Optional<Map.Entry<String, String>> mappedVersion =
+                  oid != null
+                      ? mapToFhirVersion(code.get("version"), oid, codeSystemEntries)
+                      : null;
```

Comment:
```suggestion
                  oid != null
                      ? mapToFhirVersion(code.get("version"), oid, codeSystemEntries)
                      : Optional.empty();
```
Alternatively to moving the null check inside `mapToFhirVersion`, keep with the Optional pattern and return `Optional.empty()`.

Reply:
thank you, I have made those changes